synapse/handlers/message.py

```diff
@@ -999,7 +999,14 @@ async def create_and_send_nonmember_event(
             raise ShadowBanError()
 
         if ratelimit:
-            await self.request_ratelimiter.ratelimit(requester, update=False)
+            is_admin_redaction = await self.is_admin_redaction(
+                event_type=event_dict["type"],
+                sender=event_dict["sender"],
+                redacts=event_dict.get("redacts"),
+            )
```

Comment:
Was this just a minor bug that we weren't calculating this properly for admin vs non-admin?

Reply:
yup, I think so